File: LeetCode/1351/main.py
class Solution:
    def countNegatives(self, arr: List[List[int]]) -> int:

        n = len(arr)    
        m = len(arr[0])
        row, col, ans = 0, m - 1, 0

        while(row <= n - 1 and col >= 0) :
            if(arr[row][col] < 0) :
                ans += (n - row)
                col -= 1
            else :
                row += 1
        
        return ans
    

File: LeetCode/1351/main.cpp
class Solution {
public:
  int countNegatives(vector<vector<int>> &arr) {
    int n = arr.size();
    int m = arr[0].size();

    int row = 0, col = m - 1, ans = 0;
    // if arr[row][col] < 0, then entire col < 0
    // so, we can just add (n - row) and skip the col

    while (row <= n - 1 && col >= 0) {
      int ele = arr[row][col];
      if (ele < 0) {
        ans += (n - row);
        col--;
      } else {
        row++;
      }
    }

    return ans;
  }
};